mantle/kola/tests/rhcos/luks.go

```diff
@@ -94,143 +78,101 @@ func init() {
-	server = /usr/libexec/tangdw
-	socket_type = stream
-	protocol = tcp
-	only_from = 10.0.2.15 127.0.0.1
-	user = %s
-	wait = no
-}`, cacheDir, user.Username)
-		if err := os.MkdirAll(cacheDir, 0755); err != nil {
-			c.Fatalf("Unable to create %s: %v", err)
+	ignition := conf.Ignition(`{
+		"ignition": {
+			"version": "2.2.0"
 		}
+	}`)
 
-		f, err := os.Open(cacheDir)
-		if err != nil {
-			c.Fatalf("Unable to open %s: %v", cacheDir, err)
-		}
-		// This is a simple check that the directory is empty
-		_, err = f.Readdir(1)
-		if err == io.EOF {
-			if err := os.MkdirAll(dbDir, 0755); err != nil {
-				c.Fatalf("Unable to create %s: %v", err)
-			}
-			err := ioutil.WriteFile(xinetdConfFile, []byte(xinetdConf), 0644)
-			if err != nil {
-				c.Fatalf("Unable to write xinetd configuration file: %v", err)
-			}
-			keygen := exec.Command("/usr/libexec/tangd-keygen", dbDir)
-			if err := keygen.Run(); err != nil {
-				c.Fatalf("Unable to generate Tang keys: %v", err)
-			}
-			update := exec.Command("/usr/libexec/tangd-update", dbDir, cacheDir)
-			if err := update.Run(); err != nil {
-				c.Fatalf("Unable to update Tang DB: %v", err)
+	switch pc := c.Cluster.(type) {
+	// These cases have to be separated because when put together to the same case statement
+	// the golang compiler no longer checks that the individual types in the case have the
+	// NewMachineWithOptions function, but rather whether platform.Cluster does which fails
+	case *unprivqemu.Cluster:
+		m, err = pc.NewMachineWithOptions(ignition, options, true)
+		for _, hfp := range options.HostForwardPorts {
+			if hfp.Service == "tang" {
+				tangAddress = fmt.Sprintf("10.0.2.2:%d", hfp.HostPort)
 			}
 		}
-	})
-}
-
-func getEncodedTangPin(c cluster.TestCluster) string {
-	return b64Encode(getTangPin(c))
-}
-
-func startTang(c cluster.TestCluster) int {
-	if _, err := os.Stat(xinetdPidFile); err == nil {
-		if os.Remove(xinetdPidFile); err != nil {
-			c.Fatalf("Unable to delete %s: %v", xinetdPidFile, err)
-		}
+	default:
+		m, err = pc.NewMachine(ignition)
+		tangAddress = fmt.Sprintf("%s:80", m.PrivateIP())
+	}
+	if err != nil {
+		c.Fatal(err)
 	}
 
-	cmd := exec.Command("/usr/sbin/xinetd", "-f", xinetdConfFile, "-pidfile", xinetdPidFile, "-filelog", xinetdLogFile)
-	// Give the xinetd child process a separate process group ID so it can be
-	// killed independently from the test
-	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
-	if err := cmd.Run(); err != nil {
-		c.Fatalf("Unable to start Tang: %v ", err)
+	containerID, _, err := m.SSH("sudo podman run -d -p 80:80 docker.io/mnguyenrh/tangd")
+	if err != nil {
+		c.Fatalf("Unable to start Tang container: %v", err)
 	}
 
-	// xinetd detatches itself and os.Process.Pid reports incorrect pid
-	// Use pid file instead
-	if err := util.Retry(10, 2*time.Second, func() error {
-		_, err := os.Stat(xinetdPidFile)
+	// Wait a little bit for the container to start
+	if err := util.Retry(10, time.Second, func() error {
```

Comment:
Any chance we could do a loop instead to reduce the 10s wait?  I can envision a situation where 10s is not enough, and too long.

Reply:
The `Retry` function should return early if the `func` passed in doesn't return an error.  Is this different from what you are suggesting?  There is also a `WaitUntilReady` function that takes a timeout instead of a number of attempts.